### PR TITLE
Scheduler: move more to internaltypes.ResourceList

### PR DESCRIPTION
**Pull request by robertdavidsmith**

### CODE REVIEW

File: internal/scheduler/scheduling/scheduling_algo.go [hunk 1]
```diff
@@ -182,10 +182,6 @@ type FairSchedulingAlgoContext struct {
 	Txn               *jobdb.Txn
 }
 
-func (l *FairSchedulingAlgo) NewFairSchedulingAlgoContext(ctx *armadacontext.Context, txn *jobdb.Txn, pool configuration.PoolConfig) (*FairSchedulingAlgoContext, error) {
```
Comment (robertdavidsmith) [inline internal/scheduler/scheduling/scheduling_algo.go:185]: unused

File: internal/scheduler/internaltypes/resource_list.go [hunk 2]
```diff
@@ -266,17 +304,6 @@ func (rl ResourceList) Negate() ResourceList {
 	return ResourceList{factory: rl.factory, resources: result}
 }
 
-func (rl ResourceList) Scale(factor float64) ResourceList {
```
Comment (robertdavidsmith) [inline internal/scheduler/internaltypes/resource_list.go:269]: `Scale()` is unused

File: internal/scheduler/scheduling/constraints/constraints_test.go [hunk 3]
```diff
@@ -389,98 +373,18 @@ func makeConstraintsTest(constraints SchedulingConstraints) *constraintTest {
 	}
 }
 
-func TestIsStrictlyLessOrEqual(t *testing.T) {
```
Comment (robertdavidsmith) [inline internal/scheduler/scheduling/constraints/constraints_test.go:392]: method `IsStrictlyLessOrEqual` has been deleted

File: internal/scheduler/scheduling/context/queue.go [hunk 4]
```diff
@@ -32,23 +33,20 @@ type QueueSchedulingContext struct {
...
 	// Fair share is the weight of this queue over the sum of the weights of all queues
 	FairShare float64
 	// AdjustedFairShare modifies fair share such that queues that have a demand cost less than their fair share, have their fair share reallocated.
 	AdjustedFairShare float64
 	// Total resources assigned to the queue across all clusters by priority class.
 	// Includes jobs scheduled during this invocation of the scheduler.
-	AllocatedByPriorityClass schedulerobjects.QuantityByTAndResourceType[string]
-	// Total away resources assigned to the queue across all clusters by priority class.
-	// Includes away jobs scheduled during this invocation of the scheduler.
-	AwayAllocatedByPriorityClass schedulerobjects.QuantityByTAndResourceType[string]
```
Comment (robertdavidsmith) [inline internal/scheduler/scheduling/context/queue.go:51]: AwayAllocatedByPriorityClass is unused

File: internal/scheduler/scheduling/constraints/constraints.go [hunk 5]
```diff
@@ -191,64 +139,83 @@ func (constraints *SchedulingConstraints) CheckConstraints(
...
-	return map[string]resource.Quantity{}
-}
+	rlFactory := totalResources.Factory()
 
-func (constraints *SchedulingConstraints) getPriorityClassResourceLimits(priorityClass string) map[string]resource.Quantity {
-	if priorityClassConstraint, ok := constraints.priorityClassSchedulingConstraintsByPriorityClassName[priorityClass]; ok {
-		return priorityClassConstraint.MaximumResourcesPerQueue
+	maximumResourceFractionToSchedule := config.MaximumResourceFractionToSchedule
+	if m, ok := config.MaximumResourceFractionToScheduleByPool[pool]; ok {
+		// Use pool-specific config is available.
```
Comment (JamesMurkin) [inline internal/scheduler/scheduling/constraints/constraints.go:176]:
```suggestion
		// Use pool-specific config if available.
```

**Review by JamesMurkin — APPROVED** [review]

File: internal/scheduler/scheduling/preempting_queue_scheduler_test.go [hunk 6]
```diff
@@ -2104,7 +2113,8 @@ func TestPreemptingQueueScheduler(t *testing.T) {
 					)
 				}
 				for queue, qctx := range sctx.QueueSchedulingContexts {
-					assert.True(t, qctx.AllocatedByPriorityClass.Equal(allocatedByQueueAndPriorityClass[queue]))
+					m := internaltypes.RlMapFromJobSchedulerObjects(allocatedByQueueAndPriorityClass[queue], testfixtures.TestResourceListFactory)
+					assert.Equal(t, internaltypes.RlMapRemoveZeros(m), internaltypes.RlMapRemoveZeros(qctx.AllocatedByPriorityClass))
```
Comment (robertdavidsmith) [inline internal/scheduler/scheduling/preempting_queue_scheduler_test.go:2117]: The old `schedulerobjects` `Equal` methods treat missing and zero as the same. For now, reproduce the old behavior.

File: internal/scheduler/simulator/test_utils.go [hunk 7]
```diff
@@ -88,19 +87,6 @@ func GetBasicSchedulingConfig() configuration.SchedulingConfig {
...
-	for _, cluster := range cs.Clusters {
-		for _, nt := range cluster.NodeTemplates {
-			for t, q := range nt.TotalResources.Resources {
-				total.AddQuantity(t, constraints.ScaleQuantity(q, float64(nt.Number)))
-			}
-		}
-	}
-	return total
-}
-
```
Comment (robertdavidsmith) [inline internal/scheduler/simulator/test_utils.go:103]: TotalResources is unused

**Pull request merged**